addenda99_dishonored_return.go

```diff
@@ -0,0 +1,185 @@
+	if Addenda99Dishonored == nil {
+		return
+	}
+	Addenda99Dishonored.validateOpts = opts
+}
+
+func isODFIDishonoredReturn(code string) bool {
+	switch code {
+	case "R61", "R67", "R68", "R69", "R70": // Used by ODFI for Dishonored Returns
+		return true
+	}
+	return false
+}
+
+func isRDFIDishonoredReturn(code string) bool {
+	switch code {
+	case "R71", "R72", "R73", "R74", "R75", "R76": // Used by RDFI for Dishonored Returns
+		return true
+	}
+	return false
+}
+
+func IsDishonoredReturnCode(code string) bool {
+	return isODFIDishonoredReturn(code) || isRDFIDishonoredReturn(code)
+}
+
+// Validate verifies NACHA rules for Addenda99Dishonored
+func (Addenda99Dishonored *Addenda99Dishonored) Validate() error {
+	if Addenda99Dishonored.recordType != "7" {
+		return fieldError("recordType", NewErrRecordType(7), Addenda99Dishonored.recordType)
+	}
+	if Addenda99Dishonored.TypeCode == "" {
+		return fieldError("TypeCode", ErrConstructor, Addenda99Dishonored.TypeCode)
+	}
+	if Addenda99Dishonored.TypeCode != "99" {
+		return fieldError("TypeCode", ErrAddendaTypeCode, Addenda99Dishonored.TypeCode)
+	}
+
+	// Verify the DishonoredReturnReasonCode matches expected values
+	if Addenda99Dishonored.validateOpts == nil || !Addenda99Dishonored.validateOpts.CustomReturnCodes {
+		// We can validate the Dishonored ReturnCode
+		if !IsDishonoredReturnCode(Addenda99Dishonored.DishonoredReturnReasonCode) {
+			return fieldError("DishonoredReturnReasonCode", ErrAddenda99DishonoredReturnCode, Addenda99Dishonored.DishonoredReturnReasonCode)
+		}
+	}
+
+	return nil
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) DishonoredReturnReasonCodeField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.DishonoredReturnReasonCode, 3)
+}
+
+// OriginalEntryTraceNumberField returns a zero padded TraceNumber string
+func (Addenda99Dishonored *Addenda99Dishonored) OriginalEntryTraceNumberField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.OriginalEntryTraceNumber, 15)
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) OriginalReceivingDFIIdentificationField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.OriginalReceivingDFIIdentification, 8)
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) ReturnTraceNumberField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.ReturnTraceNumber, 15)
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) ReturnSettlementDateField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.ReturnSettlementDate, 3)
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) ReturnReasonCodeField() string {
+	return Addenda99Dishonored.stringField(Addenda99Dishonored.ReturnReasonCode, 2)
+}
+
+func (Addenda99Dishonored *Addenda99Dishonored) AddendaInformationField() string {
+	return Addenda99Dishonored.alphaField(Addenda99Dishonored.AddendaInformation, 21)
+}
+
+// TraceNumberField returns a zero padded TraceNumber string
```

Comment:
I think this is really a space padded field, not zero padded (as the comment says)

Reply:
To keep the behavior matching the other Addenda99 `*Field()` calls here it'll be zero-value. The field directly will have the exact value from the source file.